### ISSUE TITLE
训练vae_ft时验证损失不会下降，模型得不到保存

### ISSUE THREAD

**Issue by cschy** [ImKeTT/pcae#7]

**Comment by ImKeTT** [comment]
Plz provide more details, e.g., training logs, detailed training command, your training data.

Maybe because of the learning rate mentioned in https://github.com/ImKeTT/PCAE/issues/5#issuecomment-1501691816 , where `--lr 1e-4` should work fine (and I've updated the readme file)

**Comment by cschy** [comment]
您好，我tokenizer使用的是BartTokenizerFast，使用facebook/bart-large，模型也是使用bart-large。训练数据使用yelp，训练命令和您readme文件中一样，具体是python train.py --run_mode vae_ft --dataset yelp --zmanner hidden --gpu 0 --dim_z 128 --per_gpu_train_batch_size 64 --train_epochs 8 --fb_mode 1 --lr 1e-4 --first_token_pooling --eval_every 500

训练过程中的验证重建损失为：
[2023-05-29 10:59:02] Val Loss Rec : 8.4928
[2023-05-29 11:03:14] Val Loss Rec : 8.7639
[2023-05-29 11:07:25] Val Loss Rec : 9.7664
[2023-05-29 11:11:33] Val Loss Rec : 11.1499
[2023-05-29 11:15:43] Val Loss Rec : 10.3339
[2023-05-29 11:19:49] Val Loss Rec : 11.4671
[2023-05-29 11:23:58] Val Loss Rec : 11.6180
[2023-05-29 11:28:05] Val Loss Rec : 9.7975
[2023-05-29 11:32:14] Val Loss Rec : 11.7892
[2023-05-29 11:36:23] Val Loss Rec : 11.7045
[2023-05-29 11:40:35] Val Loss Rec : 11.4158
[2023-05-29 11:44:43] Val Loss Rec : 11.8729
[2023-05-29 11:48:53] Val Loss Rec : 11.8641
[2023-05-29 11:53:03] Val Loss Rec : 10.5703
[2023-05-29 11:57:11] Val Loss Rec : 10.8992
[2023-05-29 12:01:21] Val Loss Rec : 11.4373
[2023-05-29 12:05:31] Val Loss Rec : 10.5476
[2023-05-29 12:09:39] Val Loss Rec : 11.7366
[2023-05-29 12:13:48] Val Loss Rec : 11.0256
[2023-05-29 12:17:57] Val Loss Rec : 11.7913
[2023-05-29 12:22:05] Val Loss Rec : 11.4671
[2023-05-29 12:26:15] Val Loss Rec : 11.2630
[2023-05-29 12:30:26] Val Loss Rec : 11.5380
[2023-05-29 12:34:33] Val Loss Rec : 11.4548
[2023-05-29 12:38:42] Val Loss Rec : 11.3381
[2023-05-29 12:42:50] Val Loss Rec : 11.0150
[2023-05-29 12:46:58] Val Loss Rec : 11.6045
[2023-05-29 12:51:05] Val Loss Rec : 11.7585
[2023-05-29 12:55:13] Val Loss Rec : 11.4332
[2023-05-29 12:59:19] Val Loss Rec : 11.1204
[2023-05-29 13:03:27] Val Loss Rec : 11.3525
[2023-05-29 13:07:35] Val Loss Rec : 12.3144
[2023-05-29 13:11:43] Val Loss Rec : 11.9962
[2023-05-29 13:15:50] Val Loss Rec : 11.4468
[2023-05-29 13:19:58] Val Loss Rec : 11.4167
[2023-05-29 13:24:05] Val Loss Rec : 11.7753
[2023-05-29 13:28:13] Val Loss Rec : 11.2144
[2023-05-29 13:32:20] Val Loss Rec : 11.5534
[2023-05-29 13:36:27] Val Loss Rec : 11.8443
[2023-05-29 13:40:36] Val Loss Rec : 12.4561
[2023-05-29 13:44:42] Val Loss Rec : 11.3503
[2023-05-29 13:48:50] Val Loss Rec : 11.1992
[2023-05-29 13:52:58] Val Loss Rec : 10.7974
[2023-05-29 13:57:06] Val Loss Rec : 11.7798
[2023-05-29 14:01:13] Val Loss Rec : 11.7107
[2023-05-29 14:05:20] Val Loss Rec : 10.8338
[2023-05-29 14:09:27] Val Loss Rec : 11.2933
[2023-05-29 14:13:34] Val Loss Rec : 11.3383
[2023-05-29 14:17:44] Val Loss Rec : 10.6144
[2023-05-29 14:21:52] Val Loss Rec : 12.5626

训练重建损失最开始7.7379，很快下降到5.6203，后面稳定在5.9左右：
[2023-05-29 14:14:07] Train Loss Rec : 5.9647
[2023-05-29 14:14:41] Train Loss Rec : 5.9625
[2023-05-29 14:15:16] Train Loss Rec : 5.9628
[2023-05-29 14:15:50] Train Loss Rec : 5.9635
[2023-05-29 14:16:24] Train Loss Rec : 5.9667
[2023-05-29 14:18:17] Train Loss Rec : 5.9597
[2023-05-29 14:18:51] Train Loss Rec : 5.9656
[2023-05-29 14:19:26] Train Loss Rec : 5.9636
[2023-05-29 14:20:00] Train Loss Rec : 5.9623
[2023-05-29 14:20:33] Train Loss Rec : 5.9607

**Comment by ImKeTT** [comment]
We only employed `bart-base` in our VAE finetuning stage, and as we have presented in the log files [here](https://drive.google.com/file/d/1hp_vm1rQIxWgCSkgm7cKtasGWxCG1kL8/view?usp=sharing), both of the training loss and evaluate loss converged during training on Yelp. 

Since you're using the `bart-large`, hyperparameters for properly training the model can be different from the `base` one. And in our experiments, learning rate (`--lr`) was a sensitive variable that should be taken care of. 

You should try different hyperparameters, like `--lr`, `--train_epochs`, `--dim_z`, for training a different (sized) model. 

Best regards.

**Comment by ImKeTT** [comment]
Do you have further questions? If yes, plz don't hesitate to let me know what I can help, or I will close this issue shortly :-)
